<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Figure 2</title>

<script src="site_libs/header-attrs-2.13/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="figure2.html">Figure 2</a>
</li>
<li>
  <a href="figure3.html">Figure 3</a>
</li>
<li>
  <a href="figure4.html">Figure 4</a>
</li>
<li>
  <a href="figure5.html">Figure 5</a>
</li>
<li>
  <a href="figure6.html">Figure 6</a>
</li>
<li>
  <a href="figure7.html">Figure 7</a>
</li>
<li>
  <a href="figure8.html">Figure 8</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Figure 2</h1>

</div>


<p>Load cohort annotation file and Epithelial cells Seurat object</p>
<div id="figure-2-a" class="section level2">
<h2>Figure 2-A</h2>
<pre class="r"><code>p4 = DimPlot(aux.seurat, 
             reduction = &quot;umap&quot;, 
             raster=FALSE, 
             group.by = &quot;Major.Class&quot;, 
             pt.size = .1, 
             split.by = &#39;Major.Class&#39;, 
             cols = c(&quot;Endometrioma&quot; = &quot;#7b3294&quot;, 
                      &quot;Eutopic Endometrium&quot; = &quot;#c2a5cf&quot;, 
                      &quot;Endometriosis&quot; = &quot;#d9f0d3&quot;, 
                      &quot;No endometriosis detected&quot; = &quot;#a6dba0&quot;,
                      &quot;Unaffected ovary&quot; = &quot;#008837&quot;))

p4</code></pre>
<p><img src="figure2_files/figure-html/unnamed-chunk-3-1.png" width="1248" style="display: block; margin: auto;" /></p>
</div>
<div id="figure-2-b" class="section level2">
<h2>Figure 2-B</h2>
<pre class="r"><code>p4 = DimPlot(object = aux.seurat, 
             pt.size = 0.1, 
             raster=FALSE, 
             label = T)
p4</code></pre>
<p><img src="figure2_files/figure-html/unnamed-chunk-4-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="figure-2-c" class="section level2">
<h2>Figure 2-C</h2>
<pre class="r"><code>p4 = DimPlot(object = aux.seurat, 
             pt.size = 0.1, 
             raster=FALSE, 
             group.by = &quot;active.cluster&quot;,
             label = T)
p4</code></pre>
<p><img src="figure2_files/figure-html/unnamed-chunk-5-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="figure-3-e" class="section level2">
<h2>Figure 3-E</h2>
<pre class="r"><code>table.gene.markers &lt;- data.frame(Gene=c(&quot;DCN&quot;, &quot;COL11A1&quot;, &quot;PDGFRA&quot;,&quot;PDGFRB&quot;), Cell=&quot;Mesenchymal cells&quot;)
table.gene.markers &lt;- rbind(table.gene.markers, data.frame(Gene=c(&quot;EPCAM&quot;, &quot;KRT8&quot;, &quot;KRT10&quot;, &quot;KRT18&quot;, &quot;KRT19&quot;), Cell=&quot;Epithelial cells&quot;))
table.gene.markers &lt;- rbind(table.gene.markers, data.frame(Gene=c(&quot;ACTA2&quot;), Cell=&quot;Smooth muscle cells&quot;))
table.gene.markers &lt;- rbind(table.gene.markers, data.frame(Gene=c(&quot;HBB&quot;, &quot;GYPA&quot;), Cell=&quot;Erythrocytes&quot;)) 
table.gene.markers &lt;- rbind(table.gene.markers, data.frame(Gene=c(&quot;CLDN5&quot;, &quot;PECAM1&quot;, &quot;CD34&quot;, &quot;ESAM&quot;), Cell=&quot;Endothelial cells&quot;))
table.gene.markers &lt;- rbind(table.gene.markers, data.frame(Gene=c(&quot;TPSB2&quot;), Cell=&quot;Mast cells&quot;))
table.gene.markers &lt;- rbind(table.gene.markers, data.frame(Gene=c(&quot;LYZ&quot;, &quot;CD14&quot;,&quot;C1QA&quot;, &quot;CLEC10A&quot;), Cell=&quot;Myeloid cells&quot;))
table.gene.markers &lt;- rbind(table.gene.markers, data.frame(Gene=c(&quot;CD2&quot;, &quot;CD3D&quot;, &quot;CD3E&quot;, &quot;CD3G&quot;, &#39;CD8A&#39;, &quot;CCL5&quot;), Cell=&quot;T/NK cells&quot;))
table.gene.markers &lt;- rbind(table.gene.markers, data.frame(Gene=c(&quot;JCHAIN&quot;, &quot;CD79A&quot;), Cell=&quot;B/Plasma cells&quot;))

p = DotPlot(object = aux.seurat, assay = &quot;RNA&quot;, group.by = &quot;active.cluster&quot;, features = table.gene.markers$Gene) +
        scale_colour_gradient2(low = &quot;#2166ac&quot;, mid = &quot;#f7f7f7&quot;, high = &quot;#b2182b&quot;) +
        RotatedAxis()
p</code></pre>
<p><img src="figure2_files/figure-html/unnamed-chunk-6-1.png" width="960" /></p>
</div>
<div id="figure-3-f" class="section level2">
<h2>Figure 3-F</h2>
<pre class="r"><code>bsize = 14

aux = data.frame(table(aux.seurat@meta.data$active.cluster))
aux = aux[order(aux$Freq),]
aux$Var1 &lt;- factor(aux$Var1, levels = aux$Var1)
order.cell = aux$Var1

pd &lt;- ggplot(aux, aes(x = Var1, y = Freq)) +
        geom_point(size=2) +
        theme_minimal(base_size = bsize) +
        #geom_text(aes(label=Freq), position=position_dodge(width=2), hjust=-0.1, size=3) +
        ylim(0, 77000) +
        scale_y_continuous(breaks=c(0, 25000, 50000, 75000, 100000, 125000)) +
        geom_hline(yintercept = c(0, 25000, 50000, 75000, 100000, 125000), linetype=&quot;dashed&quot;, color = &quot;grey&quot;) +
        #scale_x_discrete(limits = rev(levels(prop.cells$Var1))) +
        coord_flip() +
        theme(plot.title = element_text(hjust = 0.5),
              axis.text.x = element_text(angle = 270, hjust = 1),
              axis.title.x = element_blank(),
              axis.title.y = element_blank(),
              legend.position = &quot;left&quot;)</code></pre>
<pre><code>## Scale for &#39;y&#39; is already present. Adding another scale for &#39;y&#39;, which will
## replace the existing scale.</code></pre>
<pre class="r"><code>pd</code></pre>
<p><img src="figure2_files/figure-html/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>counts.prop = data.frame(table(aux.seurat@meta.data$Major.Class, aux.seurat@meta.data$active.cluster))
counts.prop.perc = group_by(counts.prop, Var2) %&gt;% mutate(percent = Freq/sum(Freq))

aux = aggregate(counts.prop$Freq, by=list(Category=counts.prop$Var1), FUN=sum)
aux$Var2 = &quot;Total&quot;
colnames(aux) &lt;- c(&quot;Var1&quot;, &quot;Freq&quot;, &quot;Var2&quot;)
aux$Var2 &lt;- factor(aux$Var2, levels = c(&quot;Total&quot;, order.cell))

aux$percent = aux$Freq / sum(aux$Freq)
counts.prop.perc = rbind(as.data.frame(counts.prop.perc), aux)

pd &lt;- ggplot(counts.prop.perc, aes(x = Var2, y = percent, fill = Var1, group=Var2)) + 
        geom_bar(position=&quot;stack&quot;,stat = &quot;identity&quot;, width=0.8) +
        scale_fill_manual(name=&quot;Class&quot;, values = c(&quot;Endometrioma&quot; = &quot;#7b3294&quot;, 
                                                   &quot;Eutopic Endometrium&quot; = &quot;#c2a5cf&quot;, 
                                                   &quot;Endometriosis&quot; = &quot;#d9f0d3&quot;, 
                                                   &quot;No endometriosis detected&quot; = &quot;#a6dba0&quot;,
                                                   &quot;Unaffected ovary&quot; = &quot;#008837&quot;)) +
        #scale_x_discrete(limits = rev(levels(counts.prop.perc$Var2))) +
        coord_flip() +
        theme_set(theme_gray(base_size = bsize)) +
        theme(plot.title = element_text(hjust = 0.5),
              axis.text.x = element_text(angle = 270, hjust = 1),
              axis.text.y = element_text(angle = 0, hjust = 1),
              axis.title.x = element_blank(),
              axis.title.y = element_blank(),
              legend.position = &quot;none&quot;)
pd</code></pre>
<p><img src="figure2_files/figure-html/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>counts.prop = data.frame(table(aux.seurat@meta.data$Patient.No., aux.seurat@meta.data$active.cluster))
counts.prop.perc = group_by(counts.prop, Var2) %&gt;% mutate(percent = Freq/sum(Freq))

col = c(&quot;#a50026&quot;, &quot;#d73027&quot;, &quot;#f46d43&quot;, &quot;#fdae61&quot;, &quot;#fee090&quot;, &quot;#ffffbf&quot;,
        &quot;#00441b&quot;, &quot;#1b7837&quot;, &quot;#5aae61&quot;, &quot;#a6dba0&quot;, &quot;#d9f0d3&quot;, &quot;#f7f7f7&quot;,
        &quot;#e0f3f8&quot;, &quot;#abd9e9&quot;, &quot;#74add1&quot;, &quot;#4575b4&quot;, &quot;#313695&quot;,
        &quot;#40004b&quot;,&quot;#762a83&quot;, &quot;#9970ab&quot;, &quot;#c2a5cf&quot;)


aux = aggregate(counts.prop$Freq, by=list(Category=counts.prop$Var1), FUN=sum)
aux$Var2 = &quot;Total&quot;
colnames(aux) &lt;- c(&quot;Var1&quot;, &quot;Freq&quot;, &quot;Var2&quot;)
aux$Var2 &lt;- factor(aux$Var2, levels = c(&quot;Total&quot;, order.cell))
aux$percent = aux$Freq / sum(aux$Freq)
counts.prop.perc = rbind(as.data.frame(counts.prop.perc), aux)



p1 &lt;- ggplot(counts.prop.perc, aes(x = Var2, y = percent, fill = Var1, group=Var2)) + 
        geom_bar(position=&quot;stack&quot;,stat = &quot;identity&quot;, width=0.8) +
        scale_fill_manual(name=&quot;Patient&quot;, values = c(  &quot;1&quot; = col[1],
                                                       &quot;2&quot; = col[2],
                                                       &quot;3&quot; = col[3],
                                                       &quot;4&quot; = col[4],
                                                       &quot;5&quot; = col[5],
                                                       &quot;6&quot; = col[6],
                                                       &quot;7&quot; = col[7],
                                                       &quot;8&quot; = col[8],
                                                       &quot;9&quot;= col[9],
                                                       &quot;10&quot; = col[10],
                                                       &quot;11&quot;= col[11],
                                                       &quot;12&quot; = col[12],
                                                       &quot;13&quot; = col[13],
                                                       &quot;14&quot; = col[14],
                                                       &quot;15&quot; = col[15],
                                                       &quot;16&quot; = col[16],
                                                       &quot;17&quot; = col[17],
                                                       &quot;18&quot; = col[18],
                                                       &quot;19&quot; = col[19],
                                                       &quot;20&quot;= col[20],
                                                       &quot;21&quot; = col[21])) +
        coord_flip() +
        theme_set(theme_gray(base_size = bsize)) +
        theme(plot.title = element_text(hjust = 0.5),
              axis.text.x = element_text(angle = 270, hjust = 1),
              axis.text.y = element_text(angle = 0, hjust = 1),
              axis.title.x = element_blank(),
              axis.title.y = element_blank(),
              legend.position = &quot;none&quot;)
p1</code></pre>
<p><img src="figure2_files/figure-html/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>aux = aux.seurat@meta.data[,c(&quot;nFeature_RNA&quot;, &quot;active.cluster&quot;)]
aux$active.cluster &lt;- factor(aux$active.cluster, levels = as.character(order.cell), ordered = TRUE)

pd &lt;- ggplot(aux, aes(y = nFeature_RNA, x = active.cluster)) +
        geom_boxplot(outlier.alpha = 0.1) + 
        theme_minimal(base_size = bsize) +
        coord_flip() + 
        #scale_x_discrete(limits = rev(levels(d10x.seurat.filt@meta.data$active.cluster))) +
        ylim(0, 10000) +
        theme(plot.title = element_text(hjust = 0.5),
              axis.text.x = element_text(angle = 270, hjust = 1),
              axis.title.x = element_blank(),
              axis.title.y = element_blank()) +
        stat_summary(fun=min, colour=&quot;darkred&quot;, geom=&quot;point&quot;, 
                     shape=18, size=3, show.legend=FALSE)
pd</code></pre>
<p><img src="figure2_files/figure-html/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="figure-3-h" class="section level2">
<h2>Figure 3-H</h2>
<pre class="r"><code>my_cluster_col = .anno[, c(&quot;Patient.No.&quot;, 
                           &quot;Major.Class&quot;, 
                           &quot;Fresh.Frozen&quot;, 
#                           &quot;Anatomic.annotation&quot;, 
                           #&quot;Pathologic.classification.&quot;, 
#                           &quot;Sequencing.batch&quot;, 
                           &quot;Menstrual.Cycle.Stage&quot;)]

#my_cluster_col$Pathologic.classification.[which(my_cluster_col$Pathologic.classification. == &quot;&quot;)] = &quot;none&quot;
#my_cluster_col$Anatomic.annotation[which(my_cluster_col$Anatomic.annotation == &quot;&quot;)] = &quot;none&quot;

my_cluster_col$ID = c(&quot;Ovarian cyst (L)&quot;,
                      &quot;Ovarian cyst (L)&quot;,
                      &quot;Ovarian cyst (R)&quot;,
                      &quot;Endometrium&quot;,
                      &quot;Anterior cul de sac&quot;,
                      &quot;Posterior cul de sac&quot;,
                      &quot;Posterior cul de sac&quot;,
                      &quot;Endometrium&quot;,
                      &quot;Ovary&quot;,
                      &quot;Pelvic side wall (R &amp; L)&quot;,
                      &quot;Anterior cul de sac&quot;,
                      &quot;Ovarian cyst (R)&quot;,
                      &quot;Uterosacral ligament (L &amp; R)&quot;,
                      &quot;Anterior cul de sac &amp; Pelvic side wall (R)&quot;,
                      &quot;Ovarian cyst (L)&quot;,
                      &quot;Uterosacral ligament (L)&quot;,
                      &quot;Rectal Serosa&quot;,
                      &quot;Pelvic side wall (R &amp; L)&quot;,
                      &quot;Ovarian fossa (L)&quot;,
                      &quot;Alterior abdominal wall &amp; Anterior pelvic cul de sac&quot;,
                      &quot;Ovarian cyst (R)&quot;,
                      &quot;Perimetrium (R)&quot;,
                      &quot;Perirectal fat&quot;,
                      &quot;Endometrium&quot;,
                      &quot;Uterosacral ligament (L)&quot;,
                      &quot;Uterosacral ligament (R)&quot;,
                      &quot;Bladder&quot;,
                      &quot;Pelvic side wall (L)&quot;,
                      &quot;Ovarian cyst (L)&quot;,
                      &quot;Endometrium&quot;, 
                      &quot;Endometrium&quot;, 
                      &quot;Endometrium&quot;, 
                      &quot;Endometriosis (R)&quot;,
                      &quot;Ovary (L)&quot;,
                      &quot;Endometrium&quot;, 
                      &quot;Ovary (R)&quot;,
                      &quot;Ovarian fossa (L)&quot;,
                      &quot;Endometrium&quot;, 
                      &quot;Endometrium&quot;, 
                      &quot;Endometrium&quot;, 
                      &quot;Ovary&quot;,
                      &quot;Ovarian cyst (L)&quot;,
                      &quot;Ovarian fossa (R)&quot;,
                      &quot;Uterosacral ligament (L)&quot;,
                      &quot;Rectal Serosa&quot;,
                      &quot;Ovarian fossa (L)&quot;,
                      &quot;Uterosacral ligament (R)&quot;,
                      &quot;Anterior cul de sac&quot;,
                      &quot;Posterior uterus&quot;
)

grid.newpage()
gridExtra::grid.table(my_cluster_col, rows = NULL)</code></pre>
<p><img src="figure2_files/figure-html/unnamed-chunk-10-1.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>aux = data.frame(table(aux.seurat@meta.data$SampleName, aux.seurat@meta.data$seurat_clusters))
data.m = spread(aux, Var2, Freq)
rownames(data.m) &lt;- data.m$Var1
data.m = t(data.m[, -1])  
data.m.prob = t(t(data.m) / colSums(data.m))

df_pca &lt;- prcomp(t(data.m.prob))
df_out &lt;- as.data.frame(df_pca$x)
df_out$Class &lt;- my_cluster_col$Major.Class
df_out$label &lt;- rownames(my_cluster_col)

percentage &lt;- round(df_pca$sdev / sum(df_pca$sdev) * 100, 2)
percentage &lt;- paste( colnames(df_out), &quot;(&quot;, paste( as.character(percentage), &quot;%&quot;, &quot;)&quot;, sep=&quot;&quot;) )

p &lt;- ggplot(data=df_out, aes(x = PC1, y = PC3, color= Class)) +
        geom_point(size = 2) +
        scale_colour_manual(name = &quot;Major Class&quot;, values =  c(&quot;Endometrioma&quot; = &quot;#7b3294&quot;, 
                                                              &quot;Eutopic Endometrium&quot; = &quot;#c2a5cf&quot;, 
                                                              &quot;Endometriosis&quot; = &quot;#d9f0d3&quot;, 
                                                              &quot;No endometriosis detected&quot; = &quot;#a6dba0&quot;,
                                                              &quot;Unaffected ovary&quot; = &quot;#008837&quot;)) +
        xlab(percentage[1]) +
        ylab(percentage[3]) +
        theme_bw(base_size = 12) +
        theme(panel.background = element_blank(),
              panel.border=element_rect(fill=NA),
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              strip.background=element_blank(),
              axis.text.x=element_text(colour=&quot;black&quot;),
              axis.text.y=element_text(colour=&quot;black&quot;),
              axis.ticks=element_line(colour=&quot;black&quot;),
              plot.margin=unit(c(1,1,1,1),&quot;line&quot;))
p</code></pre>
<p><img src="figure2_files/figure-html/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p &lt;- ggplot(data=df_out, aes(x = PC1, y = PC4, color= Class)) +
        geom_point(size = 2) +
        scale_colour_manual(name = &quot;Major Class&quot;, values =  c(&quot;Endometrioma&quot; = &quot;#7b3294&quot;, 
                                                              &quot;Eutopic Endometrium&quot; = &quot;#c2a5cf&quot;, 
                                                              &quot;Endometriosis&quot; = &quot;#d9f0d3&quot;, 
                                                              &quot;No endometriosis detected&quot; = &quot;#a6dba0&quot;,
                                                              &quot;Unaffected ovary&quot; = &quot;#008837&quot;)) +
        xlab(percentage[1]) +
        ylab(percentage[4]) +
        theme_bw(base_size = 12) +
        theme(panel.background = element_blank(),
              panel.border=element_rect(fill=NA),
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              strip.background=element_blank(),
              axis.text.x=element_text(colour=&quot;black&quot;),
              axis.text.y=element_text(colour=&quot;black&quot;),
              axis.ticks=element_line(colour=&quot;black&quot;),
              plot.margin=unit(c(1,1,1,1),&quot;line&quot;))
p</code></pre>
<p><img src="figure2_files/figure-html/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="figure-3-g" class="section level2">
<h2>Figure 3-G</h2>
<pre class="r"><code>cor.mat = cor(data.m.prob)

my_cluster_col = my_cluster_col[match(colnames(cor.mat), rownames(my_cluster_col)),]
rownames(cor.mat) &lt;- colnames(cor.mat) &lt;- paste0(my_cluster_col$Patient.No., &quot; - &quot;, my_cluster_col$ID, &quot; - &quot;, rownames(my_cluster_col))
rownames(my_cluster_col) &lt;- rownames(cor.mat)


ha2 = HeatmapAnnotation(df = my_cluster_col[, c(1, 2, 3, 4)], 
                        col = list(Major.Class = c(&quot;Endometrioma&quot; = &quot;#7b3294&quot;, 
                                                   &quot;Eutopic Endometrium&quot; = &quot;#c2a5cf&quot;, 
                                                   &quot;Endometriosis&quot; = &quot;#d9f0d3&quot;, 
                                                   &quot;No endometriosis detected&quot; = &quot;#a6dba0&quot;,
                                                   &quot;Unaffected ovary&quot; = &quot;#008837&quot;),
                                   Patient.No. = c(  &quot;1&quot; = col[1],
                                                     &quot;2&quot; = col[2],
                                                     &quot;3&quot; = col[3],
                                                     &quot;4&quot; = col[4],
                                                     &quot;5&quot; = col[5],
                                                     &quot;6&quot; = col[6],
                                                     &quot;7&quot; = col[7],
                                                     &quot;8&quot; = col[8],
                                                     &quot;9&quot;= col[9],
                                                     &quot;10&quot; = col[10],
                                                     &quot;11&quot;= col[11],
                                                     &quot;12&quot; = col[12],
                                                     &quot;13&quot; = col[13],
                                                     &quot;14&quot; = col[14],
                                                     &quot;15&quot; = col[15],
                                                     &quot;16&quot; = col[16],
                                                     &quot;17&quot; = col[17],
                                                     &quot;18&quot; = col[18],
                                                     &quot;19&quot; = col[19],
                                                     &quot;20&quot;= col[20],
                                                     &quot;21&quot; = col[21]),
                                   Fresh.Frozen = c(&quot;Fresh&quot; = &quot;#8c510a&quot;,
                                                    &quot;Frozen&quot; = &quot;#bf812d&quot;),
                                   Menstrual.Cycle.Stage = c(&quot;early-mid-late-secretory&quot; = &quot;red&quot;, 
                                                       &quot;proliferative&quot; = &quot;orange&quot;)
                        )
) 


ht1 = Heatmap(cor.mat, 
              cluster_columns = T, 
              cluster_rows = T,
              show_row_names = T, 
              row_names_gp = gpar(fontsize = 8),
              column_names_gp = gpar(fontsize = 8),
              row_names_side = &quot;right&quot;, 
              rect_gp = gpar(col = &quot;black&quot;, lwd = 0.6),
              show_column_names = T, 
              top_annotation = ha2,
              column_split = 3,
              row_split = 3,
              clustering_distance_columns = &quot;canberra&quot;,
              clustering_distance_rows = &quot;canberra&quot;,
              clustering_method_columns = &quot;ward.D2&quot;,
              clustering_method_rows = &quot;ward.D2&quot;,
              col = colorRamp2(c( min(cor.mat), mean(cor.mat), max(cor.mat) ), c(&quot;#2166ac&quot;, &quot;#f7f7f7&quot;, &quot;#b2182b&quot;))
              
)


draw(ht1, heatmap_legend_side = &quot;left&quot;, annotation_legend_side = &quot;right&quot;)</code></pre>
<p><img src="figure2_files/figure-html/unnamed-chunk-13-1.png" width="1248" style="display: block; margin: auto;" /></p>
</div>
<div id="figure-3-i" class="section level2">
<h2>Figure 3-I</h2>
<pre class="r"><code>table.class.cell = table(aux.seurat@meta.data$active.cluster, aux.seurat@meta.data$Major.Class)
table.class.cell.perc = table.class.cell/rowSums(table.class.cell)

obs.class.perc = table(aux.seurat@meta.data$Major.Class)/length(aux.seurat@meta.data$Major.Class)
table.fc = table.class.cell
table.fc = is.na(table.fc)

for (class in colnames(table.class.cell.perc)) {
        for (cell.type in rownames(table.class.cell.perc)) {
                expected = table.class.cell.perc[cell.type, class]
                obs = obs.class.perc[class]
                if (obs &gt; expected) {
                        ratio = -(obs/expected)
                } else {
                        ratio = expected/obs
                }
                #ratio = expected/obs
                
                table.fc[cell.type, class] = ratio
        }
}


quantile_breaks &lt;- function(xs, n = 10) {
        breaks &lt;- quantile(xs, probs = seq(0, 1, length.out = n))
        breaks[!duplicated(breaks)]
}
mat_breaks &lt;- quantile_breaks(table.fc, n = 11)


table.fc = table.fc[,c(&quot;Endometrioma&quot;, &quot;Endometriosis&quot;, &quot;No endometriosis detected&quot;, &quot;Eutopic Endometrium&quot;, &quot;Unaffected ovary&quot;)]

pheatmap(t(table.fc), 
         display_numbers = signif(t(table.fc), 3),
         color             = colorRampPalette(c(&quot;#81B8D5&quot;, &quot;white&quot;, &quot;#E37F65&quot;))(10),
         breaks            = mat_breaks,
         cluster_rows      = F,
         cluster_cols      = F,
         fontsize          = 14)</code></pre>
<p><img src="figure2_files/figure-html/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
