<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Figure 8</title>

<script src="site_libs/header-attrs-2.13/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="figure2.html">Figure 2</a>
</li>
<li>
  <a href="figure3.html">Figure 3</a>
</li>
<li>
  <a href="figure4.html">Figure 4</a>
</li>
<li>
  <a href="figure5.html">Figure 5</a>
</li>
<li>
  <a href="figure6.html">Figure 6</a>
</li>
<li>
  <a href="figure7.html">Figure 7</a>
</li>
<li>
  <a href="figure8.html">Figure 8</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Figure 8</h1>

</div>


<pre class="r"><code>sc = readRDS(&quot;rds/EnEpi_cells.rds&quot;)

prop.cells = data.frame(table(sc@meta.data$active.cluster))
prop.cells &lt;- prop.cells[order(prop.cells$Freq),]
prop.cells$Var1 &lt;- factor(prop.cells$Var1, levels = prop.cells$Var1)

ggplot(prop.cells, aes(Var1, Freq)) +
  geom_col() +
  theme_minimal(base_size = 15) +
  geom_text(aes(label=Freq), position=position_dodge(width=0.9), vjust=-0.25) +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.x = element_text(angle = 45, hjust = 1),
        axis.title.x = element_blank()) +
  labs(title=&quot;Number of cells in each cell type&quot;, x =&quot;Cell type&quot;, y = &quot;Number of Cells&quot;)</code></pre>
<p><img src="figure8_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<pre class="r"><code>prop.cells = data.frame(table(sc@meta.data$Major.Class))
prop.cells &lt;- prop.cells[order(prop.cells$Freq),]
prop.cells$Var1 &lt;- factor(prop.cells$Var1, levels = prop.cells$Var1)

ggplot(prop.cells, aes(Var1, Freq)) +
  geom_col() +
  theme_minimal(base_size = 15) +
  geom_text(aes(label=Freq), position=position_dodge(width=0.9), vjust=-0.25) +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.x = element_text(angle = 45, hjust = 1),
        axis.title.x = element_blank()) +
  labs(title=&quot;&quot;, x =&quot;Major class&quot;, y = &quot;Number of Cells&quot;)</code></pre>
<p><img src="figure8_files/figure-html/unnamed-chunk-2-2.png" width="672" /></p>
<pre class="r"><code>sc.sel = subset(sc, subset = Major.Class %in% c(&quot;Endometrioma&quot;, &quot;Eutopic Endometrium&quot;, &quot;Endometriosis&quot;))

single.cell.expression.set &lt;- SeuratToExpressionSet(sc.sel, delimiter=&#39;-&#39;, position=2, version=&quot;v3&quot;)
phenoData(single.cell.expression.set) &lt;- AnnotatedDataFrame(sc.sel@meta.data)


prop.cells = data.frame(table(sc.sel@meta.data$active.cluster))
prop.cells &lt;- prop.cells[order(prop.cells$Freq),]
prop.cells$Var1 &lt;- factor(prop.cells$Var1, levels = prop.cells$Var1)

ggplot(prop.cells, aes(Var1, Freq)) +
  geom_col() +
  theme_minimal(base_size = 15) +
  geom_text(aes(label=Freq), position=position_dodge(width=0.9), vjust=-0.25) +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.x = element_text(angle = 45, hjust = 1),
        axis.title.x = element_blank()) +
  labs(title=&quot;Number of cells in each cell type&quot;, x =&quot;Cell type&quot;, y = &quot;Number of Cells&quot;)</code></pre>
<p><img src="figure8_files/figure-html/unnamed-chunk-2-3.png" width="672" /></p>
<pre class="r"><code>prop.cells = data.frame(table(sc.sel@meta.data$Major.Class))
prop.cells &lt;- prop.cells[order(prop.cells$Freq),]
prop.cells$Var1 &lt;- factor(prop.cells$Var1, levels = prop.cells$Var1)

ggplot(prop.cells, aes(Var1, Freq)) +
  geom_col() +
  theme_minimal(base_size = 15) +
  geom_text(aes(label=Freq), position=position_dodge(width=0.9), vjust=-0.25) +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.x = element_text(angle = 45, hjust = 1),
        axis.title.x = element_blank()) +
  labs(title=&quot;&quot;, x =&quot;Major class&quot;, y = &quot;Number of Cells&quot;)</code></pre>
<p><img src="figure8_files/figure-html/unnamed-chunk-2-4.png" width="672" /></p>
<div id="load-gse129617-expression-data" class="section level2">
<h2>Load GSE129617 expression data</h2>
<pre class="r"><code>ensembl &lt;- useMart(&quot;ensembl&quot;, dataset=&quot;hsapiens_gene_ensembl&quot;)
bm &lt;- getBM(attributes=c(&#39;ensembl_gene_id&#39;, &#39;external_gene_name&#39;, &#39;gene_biotype&#39;, &#39;ensembl_transcript_id&#39;), mart = ensembl)


title = &quot;25 CCOC&quot;
load(&quot;data/ccoc_GSE129617.expr_25_samples.rda&quot;)
dim(GSE129617.expr)


df &lt;- data.frame(histotype = c(rep(&#39;CCOC&#39;, 25)))
rownames(df) &lt;- colnames(GSE129617.expr)
# Heapmap 0: plot top variable genes
matrix &lt;- GSE129617.expr
matrix &lt;- matrix[!duplicated(rownames(matrix)) &amp; rownames(matrix) %in% 
                   bm$external_gene_name[!is.na(bm$gene_biotype) &amp; 
                                           bm$gene_biotype == &#39;protein_coding&#39;],]
sd &lt;- apply(X = matrix, MARGIN = 1, FUN = sd)
rank &lt;- (length(sd) - rank(sd) + 1)
select &lt;- rank &lt;= 500

pheatmap(matrix[select,], cluster_rows=T, show_rownames=F, cluster_cols=T, annotation_col = df, 
         main = paste0(&#39;GSE129617 - 25 CCOC Top &#39;,sum(select),&#39; protein coding genes (n=&#39;, nrow(matrix), &quot;)&quot;))

matrix.eset = ExpressionSet(assayData=matrix, phenoData = new(&quot;AnnotatedDataFrame&quot;,data=df))</code></pre>
<div id="estimate-cell-type-proportions" class="section level3">
<h3>Estimate cell type proportions</h3>
<pre class="r"><code>Est.prop = music_prop(bulk.eset = matrix.eset,
                      sc.eset = single.cell.expression.set, 
                      clusters = &#39;active.cluster&#39;,
                      samples = &#39;SampleName&#39;,
                      verbose = F)</code></pre>
<pre class="r"><code>bp.esti.pro = read.delim(&quot;files/estimation.25-CCOC.decon.cvs&quot;, sep = &quot;,&quot;)
df &lt;- data.frame(histotype = c(rep(&#39;CCOC&#39;, 25)))
rownames(df) &lt;- rownames(bp.esti.pro)

bp.esti.pro = merge(bp.esti.pro, df, by=0)
bp.esti.pro = bp.esti.pro[match(rownames(df), bp.esti.pro$Row.names), ]</code></pre>
<pre class="r"><code>collorder = c(&quot;#8dd3c7&quot;, &quot;#ffffb3&quot;, &quot;#bebada&quot;, &quot;#fb8072&quot;, &quot;#80b1d3&quot;, &quot;#fdb462&quot;, &quot;#b3de69&quot;, &quot;#fccde5&quot;, &quot;#d9d9d9&quot;)

ht_list = rowAnnotation(text = anno_text(bp.esti.pro$Row.names, location = unit(1, &quot;npc&quot;), just = &quot;right&quot;, 
                                         gp = gpar(fontsize = 12)))

anno_width = unit(3, &quot;cm&quot;)
.cols = c(2: (ncol(bp.esti.pro)-1))

ht_list = ht_list + rowAnnotation(&quot;dist_tss&quot; = anno_barplot(bp.esti.pro[, .cols], bar_width = 1, gp = gpar(fill = collorder, fontsize = 14), 
                                          width = anno_width), show_annotation_name = FALSE) +
rowAnnotation(Histotype = anno_simple(bp.esti.pro$histotype), gp = gpar(fontsize = 20))

draw(ht_list, heatmap_legend_list = Legend(title = &quot;Epithelial clusters&quot;, labels = colnames(bp.esti.pro[, .cols]), legend_gp = gpar(fill = collorder)))</code></pre>
<p><img src="figure8_files/figure-html/unnamed-chunk-6-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="tcga-high-grade-serous-ovarian-carcinoma" class="section level2">
<h2>269 TCGA High-grade serous ovarian carcinoma</h2>
<pre class="r"><code>title = &quot;269 TCGA HGSOC&quot;

ucec.ID = subset(pd.mRNA.prim, cancer.type == &quot;OV&quot;)
dim(ucec.ID)

UCEC.RNA &lt;- TCGA.RNA[,c(1, which(colnames(TCGA.RNA) %in% ucec.ID$TCGAlong.id))]
dim(UCEC.RNA)
df &lt;- data.frame(histotype = c(rep(&#39;HGSOC&#39;, 269)))
colnames(UCEC.RNA) &lt;- gsub(&quot;^(.*?-.*?-.*?)-.*&quot;, &quot;\\1&quot;, colnames(UCEC.RNA))
rownames(df) &lt;- colnames(UCEC.RNA)[-1]

# Heapmap 0: plot top variable genes
matrix &lt;- UCEC.RNA
matrix &lt;- matrix[!duplicated(TCGA.RNA$gene_symbol) &amp; TCGA.RNA$gene_symbol %in%
                   bm$external_gene_name[!is.na(bm$gene_biotype) &amp;
                                           bm$gene_biotype == &#39;protein_coding&#39;],]
matrix[1:4, 1:4]
rownames(matrix) &lt;- matrix$gene_symbol
matrix = matrix[, -1]
matrix = matrix[complete.cases(matrix),]

sd &lt;- apply(X = matrix, MARGIN = 1, FUN = sd)
rank &lt;- (length(sd) - rank(sd) + 1)
select &lt;- rank &lt;= 500

delta1 = max(matrix, na.rm = T) - min(matrix, na.rm = T)
delta2 = max(matrix, na.rm = T) - 0
matrix = (matrix - min(matrix, na.rm = T)) * delta1 / delta2

pheatmap(log2(matrix[select,] + 1), cluster_rows=T, show_rownames=F, show_colnames = F, cluster_cols=T, annotation_col = df,
         main = paste0(title, &#39; Top &#39;,sum(select),&#39; protein coding genes (n=&#39;, nrow(matrix), &quot;)&quot;))

matrix.eset = new(&quot;ExpressionSet&quot;, exprs=as.matrix(matrix), phenoData = new(&quot;AnnotatedDataFrame&quot;,data=df))</code></pre>
<div id="estimate-cell-type-proportions-1" class="section level3">
<h3>Estimate cell type proportions</h3>
<pre class="r"><code>Est.prop = music_prop(bulk.eset = matrix.eset,
                      sc.eset = single.cell.expression.set, 
                      clusters = &#39;active.cluster&#39;,
                      samples = &#39;SampleName&#39;,
                      verbose = F)</code></pre>
<pre class="r"><code>bp.esti.pro = read.delim(&quot;files/estimation.HGSOC.decon.cvs&quot;, sep = &quot;,&quot;)
df &lt;- data.frame(histotype = c(rep(&#39;HGSOC&#39;, 269)))
rownames(df) &lt;- rownames(bp.esti.pro)

bp.esti.pro = merge(bp.esti.pro, df, by=0)
bp.esti.pro = bp.esti.pro[match(rownames(df), bp.esti.pro$Row.names), ]</code></pre>
<pre class="r"><code>collorder = c(&quot;#8dd3c7&quot;, &quot;#ffffb3&quot;, &quot;#bebada&quot;, &quot;#fb8072&quot;, &quot;#80b1d3&quot;, &quot;#fdb462&quot;, &quot;#b3de69&quot;, &quot;#fccde5&quot;, &quot;#d9d9d9&quot;)

ht_list = rowAnnotation(text = anno_text(bp.esti.pro$Row.names, location = unit(1, &quot;npc&quot;), just = &quot;right&quot;, 
                                         gp = gpar(fontsize = 12)))

anno_width = unit(3, &quot;cm&quot;)
.cols = c(2: (ncol(bp.esti.pro)-1))

ht_list = ht_list + rowAnnotation(&quot;dist_tss&quot; = anno_barplot(bp.esti.pro[, .cols], bar_width = 1, gp = gpar(fill = collorder, fontsize = 14), 
                                          width = anno_width), show_annotation_name = FALSE) +
rowAnnotation(Histotype = anno_simple(bp.esti.pro$histotype), gp = gpar(fontsize = 20))

draw(ht_list, heatmap_legend_list = Legend(title = &quot;Epithelial clusters&quot;, labels = colnames(bp.esti.pro[, .cols]), legend_gp = gpar(fill = collorder)))</code></pre>
<p><img src="figure8_files/figure-html/unnamed-chunk-10-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="gse9899---20-endometrioid" class="section level2">
<h2>GSE9899 - 20 Endometrioid</h2>
<pre class="r"><code>GSE9899.anno.clinical &lt;- read.delim(&quot;/media/Data01/public_data/CCOC/GSE9899/GSE9899_clinical_anns.csv&quot;, sep = &quot;,&quot;)
GSE9899.anno.clinical = merge(GSE9899.anno.clinical, read.delim(&quot;/media/Data01/public_data/CCOC/GSE9899/AOCSID_GSM_IDs_clinical_anns.csv&quot;, sep = &quot;,&quot;), by=&quot;AOCSID&quot;)

title = &quot;GSE9899 - 20 Endometrioid&quot;

GSE9899 &lt;- getGEO(&#39;GSE9899&#39;,GSEMatrix=TRUE)
GSE9899.expr &lt;- exprs(GSE9899$GSE9899_series_matrix.txt.gz)
head(GSE9899.expr)

annot.BM &lt;-  getBM(mart=ensembl,
                   attributes=c(&quot;affy_hg_u133_plus_2&quot;, &quot;ensembl_gene_id&quot;, &quot;gene_biotype&quot;, &quot;external_gene_name&quot;),
                   filter=&quot;affy_hg_u133_plus_2&quot;,
                   values=rownames(GSE9899.expr),
                   uniqueRows=TRUE)

common = match(rownames(GSE9899.expr), annot.BM$affy_hg_u133_plus_2)
rownames(GSE9899.expr)[!is.na(common)] = annot.BM$external_gene_name[na.omit(common)]

df &lt;- GSE9899.anno.clinical
rownames(df) &lt;- df$GSMID

df &lt;- df[order(df$Primary.Site),]

matrix &lt;- GSE9899.expr[, df$GSMID]
matrix &lt;- matrix[complete.cases(matrix),]
matrix &lt;- matrix[!duplicated(rownames(matrix)) &amp; rownames(matrix) %in%
                   bm$external_gene_name[!is.na(bm$gene_biotype) &amp;
                                           bm$gene_biotype == &#39;protein_coding&#39;],]
sd &lt;- apply(X = matrix, MARGIN = 1, FUN = sd)
rank &lt;- (length(sd) - rank(sd) + 1)
select &lt;- rank &lt;= 50
df = df[,-c(1, 7)]

df = df[df$Subtype == &quot;Endo&quot;, , drop=F]
matrix = matrix[, which(colnames(matrix) %in% rownames(df))]

df = df[,3, drop=F]
colnames(df) &lt;- &quot;histotype&quot;

matrix = matrix[-which(rownames(matrix)==&quot;&quot;),]
matrix.eset = ExpressionSet(assayData=matrix, phenoData = new(&quot;AnnotatedDataFrame&quot;,data=df))</code></pre>
<div id="estimate-cell-type-proportions-2" class="section level3">
<h3>Estimate cell type proportions</h3>
<pre class="r"><code>Est.prop = music_prop(bulk.eset = matrix.eset,
                      sc.eset = single.cell.expression.set, 
                      clusters = &#39;active.cluster&#39;,
                      samples = &#39;SampleName&#39;,
                      verbose = F)</code></pre>
<pre class="r"><code>bp.esti.pro = read.delim(&quot;files/estimation.GSE9899 - 20 Endometrioid.cvs&quot;, sep = &quot;,&quot;)
df &lt;- data.frame(histotype = c(rep(&#39;EnOC&#39;, 20)))
rownames(df) &lt;- rownames(bp.esti.pro)

bp.esti.pro = merge(bp.esti.pro, df, by=0)
bp.esti.pro = bp.esti.pro[match(rownames(df), bp.esti.pro$Row.names), ]</code></pre>
<pre class="r"><code>collorder = c(&quot;#8dd3c7&quot;, &quot;#ffffb3&quot;, &quot;#bebada&quot;, &quot;#fb8072&quot;, &quot;#80b1d3&quot;, &quot;#fdb462&quot;, &quot;#b3de69&quot;, &quot;#fccde5&quot;, &quot;#d9d9d9&quot;)

ht_list = rowAnnotation(text = anno_text(bp.esti.pro$Row.names, location = unit(1, &quot;npc&quot;), just = &quot;right&quot;, 
                                         gp = gpar(fontsize = 12)))

anno_width = unit(3, &quot;cm&quot;)
.cols = c(2: (ncol(bp.esti.pro)-1))

ht_list = ht_list + rowAnnotation(&quot;dist_tss&quot; = anno_barplot(bp.esti.pro[, .cols], bar_width = 1, gp = gpar(fill = collorder, fontsize = 14), 
                                          width = anno_width), show_annotation_name = FALSE) +
rowAnnotation(Histotype = anno_simple(bp.esti.pro$histotype), gp = gpar(fontsize = 20))

draw(ht_list, heatmap_legend_list = Legend(title = &quot;Epithelial clusters&quot;, labels = colnames(bp.esti.pro[, .cols]), legend_gp = gpar(fill = collorder)))</code></pre>
<p><img src="figure8_files/figure-html/unnamed-chunk-14-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="gse73614---ovtucc-24-ovtuendo-35" class="section level2">
<h2>GSE73614 - OvTuCC = 24 OvTuEndo = 35</h2>
<pre class="r"><code>title = &quot;GSE73614 - OvTuCC = 24 OvTuEndo = 35&quot;

five.word &lt;- function(my.string){
  elems = unlist(strsplit(as.character(my.string), &quot; &quot;))
  elems[length(elems)]
}

GSE73614.anno.clinical &lt;- read.delim(&quot;/media/Data01/public_data/CCOC/GSE73614/Sample_type.csv&quot;, sep = &quot;,&quot;)

GSE73614.anno.clinical$histotype &lt;- sapply(GSE73614.anno.clinical$histotype, five.word)
GSE73614.anno.clinical$histotype = gsub(&quot;Mayo&quot;, &quot;&quot;, GSE73614.anno.clinical$histotype)


GSE73614 &lt;- getGEO(&#39;GSE73614&#39;,GSEMatrix=TRUE)
GSE73614.expr &lt;- exprs(GSE73614$GSE73614_series_matrix.txt.gz)
head(GSE73614.expr)

GSE73614.expr = read.table(&quot;/media/Data01/public_data/CCOC/GSE73614/GSE73614_RAW.tar&quot;)

annot.BM &lt;-  getBM(mart=ensembl,
                   filters=&quot;agilent_wholegenome_4x44k_v2&quot;,
                   attributes=c(&quot;agilent_wholegenome_4x44k_v2&quot;,&quot;external_gene_name&quot;,&quot;ensembl_gene_id&quot;,&quot;gene_biotype&quot;),
                   values=rownames(GSE73614.expr),
                   uniqueRows=TRUE)
head(annot.BM)

common = match(rownames(GSE73614.expr), annot.BM$agilent_wholegenome_4x44k_v2)
rownames(GSE73614.expr)[!is.na(common)] = annot.BM$external_gene_name[na.omit(common)]
head(GSE73614.expr)


df &lt;- GSE73614.anno.clinical
rownames(df) &lt;- df$SampleName
df = df[,-1, drop=F]

matrix &lt;- GSE73614.expr

matrix = matrix[complete.cases(matrix),]
delta1 = max(matrix, na.rm = T) - min(matrix, na.rm = T)
delta2 = max(matrix, na.rm = T) - 0

matrix = (matrix - min(matrix, na.rm = T)) * delta1 / delta2
matrix[1:5, 1:4]

matrix &lt;- matrix[!duplicated(rownames(matrix)) &amp; rownames(matrix) %in% 
                   bm$external_gene_name[!is.na(bm$gene_biotype) &amp; 
                                           bm$gene_biotype == &#39;protein_coding&#39;],]
sd &lt;- apply(X = matrix, MARGIN = 1, FUN = sd)
rank &lt;- (length(sd) - rank(sd) + 1)
select &lt;- rank &lt;= 50

df = df[df$histotype %in% c(&quot;OvTuEndo&quot;, &quot;OvTuCC&quot;), , drop=F]
df = df[order(df$histotype), , drop=F]
matrix = matrix[, match(rownames(df), colnames(matrix))]

matrix = matrix[-which(rownames(matrix)==&quot;&quot;),]
matrix.eset = ExpressionSet(assayData=matrix, phenoData = new(&quot;AnnotatedDataFrame&quot;,data=df))</code></pre>
<div id="estimate-cell-type-proportions-3" class="section level3">
<h3>Estimate cell type proportions</h3>
<pre class="r"><code>Est.prop = music_prop(bulk.eset = matrix.eset,
                      sc.eset = single.cell.expression.set, 
                      clusters = &#39;active.cluster&#39;,
                      samples = &#39;SampleName&#39;,
                      verbose = F)</code></pre>
<pre class="r"><code>five.word &lt;- function(my.string){
  elems = unlist(strsplit(as.character(my.string), &quot; &quot;))
  elems[length(elems)]
}

bp.esti.pro = read.delim(&quot;files/estimation.GSE73614 - OvTuCC = 24 OvTuEndo = 35.cvs&quot;, sep = &quot;,&quot;)

GSE73614.anno.clinical &lt;- read.delim(&quot;files/Sample_type.csv&quot;, sep = &quot;,&quot;)
GSE73614.anno.clinical$histotype &lt;- sapply(GSE73614.anno.clinical$histotype, five.word)
GSE73614.anno.clinical$histotype = gsub(&quot;Mayo&quot;, &quot;&quot;, GSE73614.anno.clinical$histotype)
df &lt;- GSE73614.anno.clinical
rownames(df) &lt;- df$SampleName
df = df[,-1, drop=F]
df = df[df$histotype %in% c(&quot;OvTuEndo&quot;, &quot;OvTuCC&quot;), , drop=F]
rownames(df) &lt;- rownames(bp.esti.pro)

bp.esti.pro = merge(bp.esti.pro, df, by=0)
bp.esti.pro = bp.esti.pro[match(rownames(df), bp.esti.pro$Row.names), ]</code></pre>
<pre class="r"><code>collorder = c(&quot;#8dd3c7&quot;, &quot;#ffffb3&quot;, &quot;#bebada&quot;, &quot;#fb8072&quot;, &quot;#80b1d3&quot;, &quot;#fdb462&quot;, &quot;#b3de69&quot;, &quot;#fccde5&quot;, &quot;#d9d9d9&quot;)

ht_list = rowAnnotation(text = anno_text(bp.esti.pro$Row.names, location = unit(1, &quot;npc&quot;), just = &quot;right&quot;, 
                                         gp = gpar(fontsize = 12)))

anno_width = unit(3, &quot;cm&quot;)
.cols = c(2: (ncol(bp.esti.pro)-1))

ht_list = ht_list + rowAnnotation(&quot;dist_tss&quot; = anno_barplot(bp.esti.pro[, .cols], bar_width = 1, gp = gpar(fill = collorder, fontsize = 14), 
                                          width = anno_width), show_annotation_name = FALSE) +
rowAnnotation(Histotype = anno_simple(bp.esti.pro$histotype), gp = gpar(fontsize = 20))

draw(ht_list, heatmap_legend_list = Legend(title = &quot;Epithelial clusters&quot;, labels = colnames(bp.esti.pro[, .cols]), legend_gp = gpar(fill = collorder)))</code></pre>
<p><img src="figure8_files/figure-html/unnamed-chunk-18-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
